src/functions/avl-validate/index.ts

```diff
@@ -0,0 +1,157 @@
+import { InvalidXmlError } from "@bods-integrated-data/shared/validation";
+import { APIGatewayEvent, APIGatewayProxyResult } from "aws-lambda";
+import axios, { AxiosError } from "axios";
+import { XMLBuilder, XMLParser } from "fast-xml-parser";
+import { ZodError, z } from "zod";
+
+const requestBodySchema = z
+    .string({
+        required_error: "Body is required",
+        invalid_type_error: "Body must be a string",
+    })
+    .transform((body) => JSON.parse(body))
+    .pipe(avlValidateRequestSchema);
+
+const generateServiceRequestMessage = (currentTimestamp: string) => {
+    const serviceRequestJson = {
+        ServiceRequest: {
+            RequestTimestamp: currentTimestamp,
+            RequestorRef: "BODS",
+            VehicleMonitoringRequest: {
+                VehicleMonitoringRequest: {
+                    RequestTimestamp: currentTimestamp,
+                },
+            },
+        },
+    };
+
+    const verifiedServiceRequest = avlServiceRequestSchema.parse(serviceRequestJson);
+
+    const completeObject = {
+        "?xml": {
+            "#text": "",
+            "@_version": "1.0",
+            "@_encoding": "UTF-8",
+            "@_standalone": "yes",
+        },
+        Siri: {
+            "@_version": "2.0",
+            "@_xmlns": "http://www.siri.org.uk/siri",
+            "@_xmlns:ns2": "http://www.ifopt.org.uk/acsb",
+            "@_xmlns:ns3": "http://www.ifopt.org.uk/ifopt",
+            "@_xmlns:ns4": "http://datex2.eu/schema/2_0RC1/2_0",
+            ServiceRequest: {
+                ...verifiedServiceRequest.ServiceRequest,
+                VehicleMonitoringRequest: {
+                    ...verifiedServiceRequest.ServiceRequest.VehicleMonitoringRequest,
+                    VehicleMonitoringRequest: {
+                        "@_version": "2.0",
+                        ...verifiedServiceRequest.ServiceRequest.VehicleMonitoringRequest.VehicleMonitoringRequest,
+                    },
+                },
+            },
+        },
+    };
+
+    const builder = new XMLBuilder({
+        ignoreAttributes: false,
+        format: true,
+        attributeNamePrefix: "@_",
+    });
+
+    const request = builder.build(completeObject) as string;
+
+    return request;
+};
+
+const parseXml = (xml: string) => {
+    const parser = new XMLParser({
+        allowBooleanAttributes: true,
+        ignoreAttributes: false,
+        parseTagValue: true,
+    });
+
+    const parsedXml = parser.parse(xml) as Record<string, unknown>;
+
+    const parsedJson = avlServiceDeliverySchema.safeParse(parsedXml.Siri);
+
+    if (!parsedJson.success) {
+        logger.error(
+            "There was an error parsing the service delivery response from the data producer.",
+            parsedJson.error.format(),
+        );
+        throw new InvalidXmlError();
+    }
+
+    return parsedJson.data;
+};
+
+export const handler = async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
```

Comment:
this endpoint will need to implement the AVL producer API key check (and the appropriate secrets manager permissions set in the Terraform lambda config)